enterprise_subsidy/apps/api/v1/views/subsidy.py

```diff
@@ -90,6 +90,20 @@ def handle_exception(self, exc):
             del response.data["detail"]
         return response
 
+    def get_serializer_class(self, *args, **kwargs):
+        """
+        Return the serializer class to use for the current request.
+
+        We override the function here instead of just setting the ``serializer_class`` class attribute because that
+        attribute only supports using a single serializer for the entire viewset.  This override logic allows for the
+        serializer class to be based conditionally on the type of action.
+        """
+        if self.request.method.lower() in ('put', 'patch'):
```

Comment:
I think there's a way to get the current "action", like self.request.action or something.  that should more closely map to the method name being called.